sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2747,6 +2747,17 @@ class Dataset[T] private[sql](
     }
   }
 
+  /**
+   * Collect a Dataset as ArrowPayload byte arrays and serve to PySpark.
+   */
+  private[sql] def collectAsArrowToPython(): Int = {
+    val payloadRdd = toArrowPayloadBytes()
+    val payloadByteArrays = payloadRdd.collect()
```

Comment:
I did some experimenting with `toLocalIteratorAndServe` and did not see any performance gain - it was actually a little worse in local mode at least.  It might be worth checking out again though.  There are other ways to go about serving to python, but the current implementation seemed to be the least intrusive to Spark.